sklearn/metrics/tests/test_pairwise.py

```diff
@@ -407,6 +418,39 @@ def test_euclidean_distances():
     assert_greater(np.max(np.abs(wrong_D - D1)), .01)
 
 
+def test_masked_euclidean_distances():
+    # first check that we get right answer with missing values for X
+    X = np.array([[1.,   5.,   7.,   5.,  10.],
+                  [8., 2., 4., np.nan, 8.],
+                  [5., np.nan, 5., np.nan, 1.],
+                  [8., np.nan, np.nan, np.nan, np.nan]])
+    D1 = masked_euclidean_distances(X, missing_values="NaN")
```

Comment:
Don't bother with this. Rather, after checking that it works in the X, Y case, check that just m_e_d(X) gives the same result as m_e_d(X, X)